src/main/java/uk/gov/hmcts/reform/roleassignment/domain/service/deleteroles/DeleteRoleAssignmentOrchestrator.java

```diff
@@ -1,14 +1,177 @@
+        }
+
+        //5. persist the  requested roles  and update status
+        updateStatusAndPersist(request);
+
+        //6. check status updated by drools and take decision
+        checkAllDeleteApproved(assignmentRequest);
+
+
+        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+
+    }
+
+    private void validationByDrool(Request request, List<RequestedRole> requestedRoles) throws Exception {
+        assignmentRequest.setRequest(request);
+        assignmentRequest.setRequestedRoles(requestedRoles);
+
+        //calling drools rules for validation
+        validationModelService.validateRequest(assignmentRequest);
+
+
+    }
+
+    private void updateStatusAndPersist(Request request) {
+        for (RequestedRole requestedRole : assignmentRequest.getRequestedRoles()) {
+            requestedRole.setRequest(request);
+            if (!requestedRole.getStatus().equals(Status.APPROVED)) {
+                requestedRole.status = Status.DELETE_REJECTED;
+                requestedRole.statusSequence = Status.DELETE_REJECTED.sequence;
+            } else {
+                requestedRole.status = Status.DELETE_APPROVED;
+                requestedRole.statusSequence = Status.DELETE_APPROVED.sequence;
+            }
+            // persist history in db
+            requestEntity.getHistoryEntities().add(persistenceService.persistHistory(requestedRole, request));
+
+        }
+
+        //Persist request to update relationship with history entities
+        persistenceService.persistRequestToHistory(requestEntity);
+    }
+
+    private void checkAllDeleteApproved(AssignmentRequest parsedAssignmentRequest) {
+        // decision block
+        List<RequestedRole> deleteApprovedRoles = parsedAssignmentRequest.getRequestedRoles().stream()
+            .filter(role -> role.getStatus().equals(
+                Status.DELETE_APPROVED)).collect(
+                Collectors.toList());
+
+        if (deleteApprovedRoles.size() == parsedAssignmentRequest.getRequestedRoles().size()) {
+
+            //Delete existing Assignment records
+            deleteLiveRecords(parsedAssignmentRequest);
+
+            //insert status deleted in history table
+            insertRequestedRole(parsedAssignmentRequest, Status.DELETED);
+
+            // Update request status to approved
+            updateRequestStatus(parsedAssignmentRequest, Status.APPROVED);
+
+
+        } else {
+
+
+            //Insert requested roles  into history table with status deleted-Rejected
+            insertRequestedRole(parsedAssignmentRequest, Status.DELETE_REJECTED);
+
+            // Update request status to REJECTED
+            updateRequestStatus(parsedAssignmentRequest, Status.REJECTED);
+
+        }
+    }
+
+    private void deleteLiveRecords(AssignmentRequest assignmentRequest) {
+        for (RoleAssignment requestedRole : assignmentRequest.getRequestedRoles()) {
+            persistenceService.deleteRoleAssignmentByActorId(requestedRole.getActorId());
```

Comment:
we can't delete by Actor if request is for process+reference

Reply:
Done.